src/sugar.jl

```diff
@@ -1159,3 +1159,75 @@ grad
     )
     return nothing
 end
+
+"""
+    autodiff(
+        rmode::Union{ReverseMode,ReverseModeSplit},
+        f::Annotation,
+        dresult::Seed,
+        annotated_args...
+    )
+    
+Call [`autodiff_thunk`](@ref) in split mode, execute the forward pass, increment output adjoint with `dresult`, then execute the reverse pass.
+
+Useful for computing pullbacks / VJPs for functions whose output is not a scalar, or when the scalar seed is not 1.
+"""
+function autodiff(
+        rmode::Union{ReverseMode{ReturnPrimal}, ReverseModeSplit{ReturnPrimal}},
+        f::FA,
```

Comment:
Okay I can get behind the Seed being explicit (sort of as an alternative to the return activity specifier).

This seems like a good/reasonable design to me, but since it's a nontrivial API design question I'm wondering if you thought about alternatives/implications

Reply:
in particular I really like how the seed can still be a part of "normal" autodiff, rather than requriing a separate autodiff_seeded, etc fn

Reply:
We thought about it a bit with Valentin [earlier in the PR](https://github.com/EnzymeAD/Enzyme.jl/pull/2309#issuecomment-2662617902). I don't see any obvious alternatives besides a separate name for the function, which is less esthetically pleasant.